src/test/regress/sql/offload_entry_to_qe.sql

```diff
@@ -0,0 +1,159 @@
+select count(x) over () from tst;
+select count(x) over () from tst group by x limit 1;
+select distinct min(x), max(x) from tst;
+set enable_offload_entry_to_qe = on;
+select count(x) from tst where x > 1;
+select count(x) from tst;
+select count(x) from tst limit 1;
+select count(x) over () from tst;
+select count(x) over () from tst group by x limit 1;
+select distinct min(x), max(x) from tst;
+
+-- accept Merge Join, Nested Loop and Hash Join
+set enable_seqscan = off;
+set enable_nestloop = off;
+set enable_mergejoin = on;
+explain (costs off, locus) select * from tst t1, tst t2 where t1.y = t2.y order by t1.y;
+set enable_offload_entry_to_qe = off;
+select * from tst t1, tst t2 where t1.y = t2.y order by t1.y;
+set enable_offload_entry_to_qe = on;
+select * from tst t1, tst t2 where t1.y = t2.y order by t1.y;
+reset enable_mergejoin;
+reset enable_nestloop;
+reset enable_seqscan;
+explain (costs off, locus) select * from (select min(x) x from tst) t1 join (values (1), (10)) v(x) on true;
+explain (costs off, locus) select * from (select min(x) x from tst) t1 join (values (1), (10)) v(x) on t1.x = v.x;
+set enable_offload_entry_to_qe = off;
+select * from (select min(x) x from tst) t1 join (values (1), (10)) v(x) on true;
+select * from (select min(x) x from tst) t1 join (values (1), (10)) v(x) on t1.x = v.x;
+set enable_offload_entry_to_qe = on;
+select * from (select min(x) x from tst) t1 join (values (1), (10)) v(x) on true;
+select * from (select min(x) x from tst) t1 join (values (1), (10)) v(x) on t1.x = v.x;
+
+-- accept InitPlan and SubPlan
+explain (costs off, locus) select count(*) from tst where tst.y = (select max(y) from tst);
+explain (costs off, locus) select (select max((select distinct x from tst t2 where t2.x = t1.x))) from tst t1;
+set enable_offload_entry_to_qe = off;
+select count(*) from tst where tst.y = (select max(y) from tst);
+select (select max((select distinct x from tst t2 where t2.x = t1.x))) from tst t1;
+set enable_offload_entry_to_qe = on;
+select count(*) from tst where tst.y = (select max(y) from tst);
+select (select max((select distinct x from tst t2 where t2.x = t1.x))) from tst t1;
+
+-- test cte
+explain (costs off, locus)
+with t(a,b) as (select min(y), max(y) from tst),
+     unused(a) as (select * from tst)
+select t.b, rank() over () from t; -- test Subquery on QD
+
+explain (costs off, locus)
+with t1(a,b) as (select min(y), max(y) from tst),
+     t2(a,b) as (select x, count(x) over () from tst group by x)
+select * from t1 join t2 on t1.a < t2.a order by t1.a, t2.a;
+
+set enable_offload_entry_to_qe = off;
+with t(a,b) as (select min(y), max(y) from tst),
+     unused(a) as (select * from tst)
+select t.b, rank() over () from t;
+with t1(a,b) as (select min(y), max(y) from tst),
+     t2(a,b) as (select x, count(x) over () from tst group by x)
+select * from t1 join t2 on t1.a < t2.a order by t1.a, t2.a;
+
+set enable_offload_entry_to_qe = on;
+with t(a,b) as (select min(y), max(y) from tst),
```

Comment:
plus RECURSIVE cte.

Reply:
I think recursive CTE already supported:
```sql
create table t_rep_test_rcte(c int) distributed replicated;
create table t_rand_test_rcte(c int) distributed by (c);
insert into t_rep_test_rcte values (1);
insert into t_rand_test_rcte values (1), (2), (3);
analyze t_rep_test_rcte;
analyze t_rand_test_rcte;

explain (slicetable, costs off, locus) with recursive the_cte_here(n) as (
  select * from t_rep_test_rcte
  union all
  select n+1 from the_cte_here join t_rand_test_rcte
                      on t_rand_test_rcte.c = the_cte_here.n)
select avg(n) over (partition by n) from the_cte_here;
                                  QUERY PLAN
------------------------------------------------------------------------------
 Gather Motion 1:1  (slice1; segments: 1)
   Locus: Entry
   ->  WindowAgg
         Locus: SingleQE
         Partition By: t_rep_test_rcte.c
         ->  Sort
               Locus: SingleQE
               Sort Key: t_rep_test_rcte.c
               ->  Recursive Union
                     Locus: SingleQE
                     ->  Gather Motion 1:1  (slice2; segments: 1)
                           Locus: SingleQE
                           ->  Seq Scan on t_rep_test_rcte
                                 Locus: SegmentGeneral
                     ->  Hash Join
                           Locus: SingleQE
                           Hash Cond: (the_cte_here.n = t_rand_test_rcte.c)
                           ->  WorkTable Scan on the_cte_here
                                 Locus: SingleQE
                           ->  Hash
                                 Locus: SingleQE
                                 ->  Gather Motion 3:1  (slice3; segments: 3)
                                       Locus: SingleQE
                                       ->  Seq Scan on t_rand_test_rcte
                                             Locus: Hashed
 Slice 0: Dispatcher; root 0; parent -1; gang size 0
 Slice 1: Singleton Reader; root 0; parent 0; gang size 1; segment 2
 Slice 2: Singleton Reader; root 0; parent 1; gang size 1; segment 2
 Slice 3: Reader; root 0; parent 1; gang size 3
 Optimizer: Postgres query optimizer
(30 rows)
```
Rightside of RecursiveUnion is a SingleQE even doesn't enable offloading, and the left side will be taken care of just as `Limit`: when there's extra computation node on Entry _besides `RecursiveUnion`, it'll be offloaded as intended. but if there's only a `RecursiveUnion` on Entry then offloading won't be triggered.

Reply:
fixed in https://github.com/cloudberrydb/cloudberrydb/compare/1a9288305477b2e647d55889c1a374121f5a62f9..c5cb4c91d8a12fdd580a0f96553ab38e63da4df4.